tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -1330,6 +1330,151 @@ def assert_shape_match(inp, out):
     return (res_outputs, new_state)
 
 
+class LayerNormBasicGRUCell(rnn_cell_impl._LayerRNNCell):
+  """GRU unit with layer normalization.
+
+    This class adds layer normalization to a
+    basic GRU unit. Layer normalization implementation is based on:
+
+      https://arxiv.org/abs/1607.06450.
+
+    "Layer Normalization"
+    Jimmy Lei Ba, Jamie Ryan Kiros, Geoffrey E. Hinton
+
+    and is applied before the internal nonlinearities.
+    """
+
+  def __init__(self, num_units,
+               activation=math_ops.tanh,
+               layer_norm=True,
+               norm_gain=1.0,
+               norm_shift=0.0,
+               reuse=None):
+    """Initializes the cell.
+
+        Args:
+          num_units: int, The number of units in the GRU cell.
+          activation: Activation function of the inner states.
+          layer_norm: If `True`, layer normalization will be applied.
+          norm_gain: float, The layer normalization gain initial value. If
+            `layer_norm` has been set to `False`, this argument will be ignored.
+          norm_shift: float, The layer normalization shift initial value. If
+            `layer_norm` has been set to `False`, this argument will be ignored.
+        """
+
+    super(LayerNormBasicGRUCell, self).__init__(_reuse=reuse)
+
+    self._num_units = num_units
+    self._activation = activation
+    self._layer_norm = layer_norm
+    self._g = norm_gain
+    self._b = norm_shift
+    self._reuse = reuse
+
+  @property
+  def state_size(self):
+    return self._num_units
+
+  @property
+  def output_size(self):
+    return self._num_units
+
+  def _linear(self, args, kernel, bias):
+    out = math_ops.matmul(args, kernel)
+    if not self._layer_norm:
+      out = nn_ops.bias_add(out, bias)
+    return out
+
+  def _norm(self, inp, scope):
+    return layers.layer_norm(inp, reuse=True, scope=scope)
```

Comment:
add some extra documentation describing that the gamma and beta variables used by this layer_norm come from the ones you built in `build`.  is there a way to pass them in explicitly instead?  that would be cleaner.

btw, are you sure you set the `begin_norm_axis` and `begin_params_axis` correctly to [layer_norm](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/layers.py#L2058)?

Reply:
Yes I agree this is kind of a hacky solution. I did this to be consistent with what has been done for LSTM. I think if we want alpha/beta initialization to be parametrizable, <a href=https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/layers/python/layers/layers.py#L2058>layer_norm</a> should probably take initializers in parameters, or at least float values for alpha/beta to be used in constant initializer. This is probably out of scope of this PR, but if you think this is worth it I can keep it for another PR. For now I'll add some extra documentation.
For `begin_norm_axis` and `begin_params_axis`, default values, respectively 1 and -1,  already match the expected behavior.